File: src/falcon_core/math/labelled_arrays/labelled_measured_arrays.py
"""A group of arrays that have been labelled and kept together."""

from .base_labelled_arrays import BaseLabelledArrays
from .dependancies import Generic, TypeVar, arrayND
from .labelled_measured_array import LabelledMeasuredArray

U = TypeVar("U", bound=arrayND)


class LabelledMeasuredArrays(BaseLabelledArrays[LabelledMeasuredArray[U]], Generic[U]):
    """A series of arrays that have been labelled and kept together."""

    def __init__(self, arrays: list[LabelledMeasuredArray[U]]) -> None:
        """Initialize the LabelledArrays object."""
        super().__init__(arrays=arrays)

File: src/math/arrays/LabelledMeasuredArrays.cpp
#include "falcon_core/math/arrays/LabelledMeasuredArrays.hpp"

namespace falcon_core::math::arrays {

LabelledMeasuredArrays::LabelledMeasuredArrays() = default;
LabelledMeasuredArrays::LabelledMeasuredArrays(
    const std::vector<LabelledMeasuredArraySP>& arrays)
    : BaseLabelledArrays(arrays) {}
}  // namespace falcon_core::math::arrays

CEREAL_REGISTER_TYPE(falcon_core::math::arrays::LabelledMeasuredArrays)
CEREAL_REGISTER_POLYMORPHIC_RELATION(
    falcon_core::math::arrays::BaseLabelledArrays<
        falcon_core::math::arrays::LabelledMeasuredArray>,
    falcon_core::math::arrays::LabelledMeasuredArrays)
